### PR TITLE
Added Go app to stream the compress and upload rather than hitting the disk

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by carlosroman — APPROVED** [review]
I assume tests coming later ;)

File: app.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+	bw, err := b.PutWriter(path, nil, nil)
+	if err != nil {
+		return err
+	}
+	defer bw.Close()
+
+	sw := snappy.NewBufferedWriter(bw)
+	tw := tar.NewWriter(sw)
+	defer sw.Close()
+	defer tw.Close()
```
Comment (carlosroman) [inline app.go:163]: Do you need to defer both or does closing the outer one close the inner one? Does one have to happen before the other or will it panic if `sw` closes before `tw`. If so shouldn't they be deferred in an anonymous function to guarantee order.

File: app.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+		return err
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func formattedNow() string {
+	return time.Now().UTC().Format("2006-01-02T15-04-05")
```
Comment (carlosroman) [inline app.go:199]: This always returns `2009-11-10T23-00-00` for me. Is that what was intended?

File: app.go (hunk 1 above)
Comment (ghost) [reply app.go:163]: They both need to be deferred, and in that order, else the resultant file never writes it's full size properly.

File: app.go (hunk 2 above)
Comment (ghost) [reply app.go:199]: That's the playground 😆

Comment (carlosroman) [reply app.go:199]: Ugh... I hate go date formats...

File: app.go (hunk 1 above)
Comment (kghatton) [reply app.go:163]: @carlosroman defer is LIFO, so tw will be closed first.

Comment (carlosroman) [reply app.go:163]: Yeah, just read up on my Go... this is what happens when I "play" with JavaScript and Python too long.

Comment (carlosroman) [reply app.go:163]: Stop writing Go for four months and I have to re-read the little Go handbook.

**Pull request merged**